src/flag_gems/ops/kron.py

```diff
@@ -0,0 +1,161 @@
+        out_indices.insert(0, remaining % dim_size)
+        remaining //= dim_size
+
+    a_idx = 0
+    b_idx = 0
+    for out_idx, (a_dim, b_dim) in zip(out_indices, zip(a_batch_dims, b_batch_dims)):
+        a_idx = a_idx * a_dim + out_idx // b_dim
+        b_idx = b_idx * b_dim + out_idx % b_dim
+
+    return a_idx, b_idx
+
+
+@triton.autotune(configs=runtime.get_tuned_config("kron"), key=["M", "N"])
+@triton.jit
+def kron_kernel(
+    a_ptr,
+    b_ptr,
+    c_ptr,
+    M: tl.int64,
+    N: tl.int64,
+    M1: tl.int64,
+    M2: tl.int64,
+    N1: tl.int64,
+    N2: tl.int64,
+    a_stride_0: tl.int64,
+    a_stride_1: tl.int64,
+    b_stride_0: tl.int64,
+    b_stride_1: tl.int64,
+    c_stride_0: tl.int64,
+    c_stride_1: tl.int64,
+    BLOCK_M: tl.constexpr,
+    BLOCK_N: tl.constexpr,
+):
+    pid = tle.program_id(0).to(tl.int64)
+    grid_n = tl.cdiv(N, BLOCK_N).to(tl.int64)
+
+    pid_m = (pid // grid_n).to(tl.int64)
+    pid_n = (pid % grid_n).to(tl.int64)
+
+    offs_am = (pid_m * BLOCK_M + tl.arange(0, BLOCK_M)).to(tl.int64)
+    offs_an = (pid_n * BLOCK_N + tl.arange(0, BLOCK_N)).to(tl.int64)
+
+    mask = (offs_am[:, None] < M) & (offs_an[None, :] < N)
+
+    a_row = (offs_am[:, None] // M2).to(tl.int64)
+    a_col = (offs_an[None, :] // N2).to(tl.int64)
+    b_row = (offs_am[:, None] % M2).to(tl.int64)
+    b_col = (offs_an[None, :] % N2).to(tl.int64)
+
+    a_idx = (a_row * a_stride_0 + a_col * a_stride_1).to(tl.int64)
+    b_idx = (b_row * b_stride_0 + b_col * b_stride_1).to(tl.int64)
+
+    a = tl.load(a_ptr + a_idx, mask=mask)
+    b = tl.load(b_ptr + b_idx, mask=mask)
+
+    c = a * b
+
+    c_idx = (offs_am[:, None] * c_stride_0 + offs_an[None, :] * c_stride_1).to(tl.int64)
+    tl.store(c_ptr + c_idx, c, mask=mask)
+
+
+def kron(A, B):
+    if A.numel() == 0 or B.numel() == 0:
+        _, _, out_shape = prepare_tensor_for_kron(A, B)
+        return torch.empty(out_shape, device=A.device, dtype=A.dtype)
+
+    if A.dim() == 0 and B.dim() == 0:
+        return A * B
+
+    A_prepared, B_prepared, out_shape = prepare_tensor_for_kron(A, B)
+
+    if not out_shape:
+        return torch.empty(0, device=A.device, dtype=A.dtype)
+
+    M1, N1 = A_prepared.shape[-2:]
+    M2, N2 = B_prepared.shape[-2:]
+    M, N = M1 * M2, N1 * N2
+
+    batch_size = int(torch.prod(torch.tensor(out_shape[:-2]))) if out_shape[:-2] else 1
+
+    C = torch.empty(out_shape, device=A.device, dtype=A.dtype)
+    C_reshaped = C.view(-1, M, N)
+
+    A_view = A_prepared.reshape(-1, M1, N1)
+    B_view = B_prepared.reshape(-1, M2, N2)
+
+    grid = lambda meta: (
+        triton.cdiv(M, meta["BLOCK_M"]) * triton.cdiv(N, meta["BLOCK_N"]),
+    )
+
+    if not A_view.is_contiguous():
+        A_view = A_view.contiguous()
+    if not B_view.is_contiguous():
+        B_view = B_view.contiguous()
+
+    for i in range(batch_size):
+        a_idx, b_idx = calculate_batch_indices(i, A_prepared.shape, B_prepared.shape)
+        kron_kernel[grid](
```

Comment:
You need to specify onto which device the kernel is going to be launched, or else it always launched onto device 0.